tests/components/config/test_device_registry.py

```diff
@@ -126,3 +132,239 @@ async def test_update_device(hass, client, registry, payload_key, payload_value)
+            "id": 5,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": fake_entry_id,
+            "device_id": device_entry.id,
+        }
+    )
+    response = await ws_client.receive_json()
+    assert response["success"]
+    assert set(response["result"]["config_entries"]) == {
+        entry_1.entry_id,
+        entry_2.entry_id,
+        entry_3.entry_id,
+    }
+
+    # Try removing a config entry which does not support removal from the device
+    await ws_client.send_json(
+        {
+            "id": 6,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": entry_1.entry_id,
+            "device_id": device_entry.id,
+        }
+    )
+    response = await ws_client.receive_json()
+
+    assert not response["success"]
+    assert response["error"]["code"] == "home_assistant_error"
+    assert (
+        response["error"]["message"] == "Config entry does not support device removal"
+    )
+
+    # Try removing a config entry from a device which does not exist
+    await ws_client.send_json(
+        {
+            "id": 7,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": entry_2.entry_id,
+            "device_id": fake_device_id,
+        }
+    )
+    response = await ws_client.receive_json()
+
+    assert not response["success"]
+    assert response["error"]["code"] == "home_assistant_error"
+    assert response["error"]["message"] == "Unknown device"
+
+    # Try removing a config entry from a device which it's not connected to - does nothing
+    await ws_client.send_json(
+        {
+            "id": 8,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": entry_2.entry_id,
+            "device_id": device_entry.id,
+        }
+    )
+    response = await ws_client.receive_json()
+
+    assert response["success"]
+    assert set(response["result"]["config_entries"]) == {
+        entry_1.entry_id,
+        entry_3.entry_id,
+    }
+
+    await ws_client.send_json(
+        {
+            "id": 9,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": entry_2.entry_id,
+            "device_id": device_entry.id,
+        }
+    )
+    response = await ws_client.receive_json()
+
+    assert response["success"]
+    assert set(response["result"]["config_entries"]) == {
+        entry_1.entry_id,
+        entry_3.entry_id,
+    }
+
+    # Try removing a config entry which can't be loaded from a device - allowed
+    await ws_client.send_json(
+        {
+            "id": 10,
+            "type": "config/device_registry/remove_config_entry",
+            "config_entry_id": entry_3.entry_id,
+            "device_id": device_entry.id,
+        }
+    )
+    response = await ws_client.receive_json()
+
+    assert response["success"]
+    assert set(response["result"]["config_entries"]) == {entry_1.entry_id}
```

Comment:
This tests the case when the config entry is present, but the integration is not loaded.